### PR TITLE
FINERACT-1958: Rework down-payment calculation

### PR DESCRIPTION
**Pull request by adamsaghy**
## Description

**2 tests got disabled:**
LoanAccountPaymentAllocationWithOverlappingDownPaymentInstallmentTest
1. loanAccountWithEnableDownPaymentAndDisableAutoRepaymentForDownPaymentWithOverlappingInstallmentPaymentAllocationTest()
2. loanAccountWithEnableDownPaymentAndEnableAutoRepaymentForDownPaymentWithOverlappingInstallmentPaymentAllocationTest()

**A follow up PR will fix them!**

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java [hunk 1]
```diff
@@ -137,18 +137,27 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
...
                 /* fetches the first tranche amount and also updates other tranche details to map */
-                BigDecimal disburseAmt = getDisbursementAmount(loanApplicationTerms, scheduleParams.getPeriodStartDate(), periods,
-                        chargesDueAtTimeOfDisbursement, scheduleParams.getDisburseDetailMap(), scheduleParams.applyInterestRecalculation(),
-                        scheduleParams);
-                scheduleParams.setPrincipalToBeScheduled(Money.of(currency, disburseAmt));
-                loanApplicationTerms.setPrincipal(loanApplicationTerms.getPrincipal().zero().plus(disburseAmt));
-                scheduleParams.setOutstandingBalanceAsPerRest(Money.of(currency, disburseAmt));
+                BigDecimal disburseAmt = getDisbursementAmount(loanApplicationTerms, scheduleParams.getPeriodStartDate(),
+                        scheduleParams.getDisburseDetailMap(), scheduleParams.applyInterestRecalculation());
+                BigDecimal downPaymentAmt = MathUtil.percentageOf(disburseAmt,
```
Comment (ruchiD) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java]: Check for downpayment enable is needed here.

**Review by ruchiD — COMMENTED** [review]
1.Not clear how we are handling overlapping downpayment installments during generate flow.
2.Not clear how we are handling Number of Repayments check during schedule generation.

**Comment by adamsaghy** [comment]
> 1.Not clear how we are handling overlapping downpayment installments during generate flow. 2.Not clear how we are handling Number of Repayments check during schedule generation.

1. It does not. Please have a follow up PR to handle them properly
2. The number of repayments does not contain the down-payment installment anymore. We need to only set how many regular installments we want.

**Review by ruchiD — COMMENTED** [review]
LGTM

**Pull request merged**